exchanges/request/request.go

```diff
@@ -174,7 +174,8 @@ func (r *Requester) doRequest(req *http.Request, p *Item) error {
 				delay = after
 			}
 
-			if d, ok := req.Context().Deadline(); ok && d.After(time.Now().Add(delay)) {
+			// what to do if req.Context().Deadline() is 0 ? Because it never passed here...
```

Comment:
We have a default retry policy that should stop this from continuously retrying. Are you having issues here? There are 10 exchanges that utilise context.WithDeadline() functionality.

Reply:
Yes, on Binance I ran into a problem when a lot of requests were sent to the server, the server banned me and returned a funny message "I'm a teapot". And, due to the fact that in this place we do not receive context.WithDeadline () - the gocryptortader was frozen for 20 minutes, and it was not possible to stop it using CTRL+C, only via KILL -9.  I have not yet figured out the issue, why the retry policy did not work. But on Binance this happened often, during orderbooks synchronization for websockets (via rest) on a large number of pairs

Reply:
Are you by any chance running multiple instances of GoCryptoTrader? Our rate limits should prevent the teapot error from ever occurring along with the banning from Binance's side. If the rate limits are inaccurate, we could re-assess them and potentially add a context.Deadline for Binance in another PR.

Reply:
No, I do not run a multiple instances, And I still don't understand why the requests limiter did not work.

Reply:
Please remove comment

Reply:
done